thirdeye/thirdeye-frontend/app/pods/components/rootcause-select-metric-dimension/component.js

```diff
@@ -0,0 +1,115 @@
+import Ember from 'ember';
+import fetch from 'fetch';
+import { filterPrefix, checkStatus, toFilters, stripTail, toFilterMap, appendFilters, fromFilterMap, toBaselineUrn, toCurrentUrn } from 'thirdeye-frontend/helpers/utils';
+import _ from 'lodash'
+
+export default Ember.Component.extend({
+  selectedUrn: null,
+
+  onSelection: null,
+
+  //
+  // internal
+  //
+  baseUrn: null, // ""
+
+  baseUrnCache: null, // ""
+
+  filterMap: {},
+
+  filterOptions: {},
+
+  didReceiveAttrs() {
+    this._super(...arguments);
+
+    const { selectedUrn, baseUrnCache } = this.getProperties('selectedUrn', 'baseUrnCache');
+
+    if (!selectedUrn || !selectedUrn.startsWith('thirdeye:metric:')) {
+      this.setProperties({
+        baseUrn: null,
+        filterMap: {},
+        filterOptions: {}
+      });
+      return;
+    }
+
+    const baseUrn = stripTail(selectedUrn);
+    const filterMap = toFilterMap(toFilters(selectedUrn));
+
+    if (!_.isEqual(baseUrn, baseUrnCache)) {
+      this._fetchFilters(baseUrn);
+      this.setProperties({ baseUrnCache: baseUrn });
+    }
+
+    this.setProperties({ baseUrn, filterMap });
+  },
+
+  filters: Ember.computed('filterMap', {
+    get() {
+      const { filterMap } = this.getProperties('filterMap');
+      return JSON.stringify(filterMap);
+    },
+    set() {
```

Comment:
delete if not needed

Reply:
required, otherwise changing the filter will permanently override the CP